scripts/prepare_for_nuget_pack.sh

```diff
@@ -0,0 +1,69 @@
+#!/bin/bash
+
+# NuGet packing doesn't support symlinks, so zip up our frameworks first to preserve the symlinks
+
+# Make a directory if necessary, no-op if it already exists
+#
+# \param $1 the name of the folder to create
+function make_dir_if_necessary()
+{
+    if ! [ -d "$1" ]; then
+        echo "Making $1 folder"
+        mkdir "$1"
+    else
+        echo "$1 folder already exists"
+    fi
+}
+
+# Build up our output directory in Products/nuget
+PRODUCTS_DIR="DerivedData/Build/Products"
+NUGET_OUTPUT_DIR="$PRODUCTS_DIR/nuget"
+make_dir_if_necessary "$NUGET_OUTPUT_DIR"
+
+NUGET_OUTPUT_INCLUDE_DIR="$NUGET_OUTPUT_DIR/include"
+make_dir_if_necessary "$NUGET_OUTPUT_INCLUDE_DIR"
+
+NUGET_OUTPUT_INCLUDE_DIR_IOS="$NUGET_OUTPUT_INCLUDE_DIR/ios"
+make_dir_if_necessary "$NUGET_OUTPUT_INCLUDE_DIR_IOS"
+
+# Copy a single generated header into our output directory in an includes folder. Pick the release device header since that's likely the most important target
+rsync -a "DerivedData/Build/Intermediates.noindex/FluentUI.build/Release-iphoneos/FluentUILib.build/DerivedSources/FluentUILib-Swift.h" "$NUGET_OUTPUT_INCLUDE_DIR_IOS/FluentUILib-Swift.h"
+
+# cd into the products directory to make copying all the output easier
+cd $PRODUCTS_DIR
+
+# Copy each platform. Note that we're omitting binary swiftmodule files in favor of swiftinterface files.
+make_dir_if_necessary "nuget/Debug-macosx"
+rsync -a --exclude '*.swiftsourceinfo' --exclude '*.swiftdoc' --exclude '*.swiftmodule' Debug/FluentUI.framework/ nuget/Debug-macosx/FluentUI.framework/
+
+make_dir_if_necessary "nuget/Ship-macosx"
+rsync -a --exclude '*.swiftsourceinfo' --exclude '*.swiftdoc' --exclude '*.swiftmodule' Release/FluentUI.framework/ nuget/Ship-macosx/FluentUI.framework/
+
+make_dir_if_necessary "nuget/Debug-iphoneos"
+rsync -a Debug-iphoneos/libFluentUILib.a nuget/Debug-iphoneos/
+rsync -a --exclude 'Project/' --exclude '*.swiftdoc' --exclude '*.swiftmodule' Debug-iphoneos/FluentUILib.swiftmodule/ nuget/Debug-iphoneos/FluentUILib.swiftmodule/
```

Comment:
the --excludes seem heavily repeated. Is there a way to factor out?

Reply:
This has been extracted into a bash variable.